### PR TITLE
Create supported ruby versions action

### PR DESCRIPTION
**Pull request by dcaddell**
Creates a Github Action that we can use to generate Ruby version arrays for use as a matrix strategy.

See `.github/workflows/test.yml` and [the resulting action run](https://github.com/Invoca/supported_ruby_versions_action/actions/runs/10726271597) for a demo.

### CODE REVIEW

**Review by jebentier — APPROVED** [review]
Nicely done!

File: .github/workflows/update_versions.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  schedule:
+  - cron: '0 0 2 1 *' # Run at midnight on January 2nd to pick up Christmas Ruby releases after the holiday break
+jobs:
+  update-versions:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - run: |
+        ./bin/update_version_cache
+        if [ -n "$(git status --porcelain)" ]; then
```
Comment (yinonrousso) [inline .github/workflows/update_versions.yml:14]: This is a cool git command! I've never seen it until now 🌟

**Review by yinonrousso — APPROVED** [review]
Great work putting this all together! It's looking quite clean ✨

File: .github/workflows/update_versions.yml (hunk 1 above)
Comment (dcaddell) [reply .github/workflows/update_versions.yml:14]: I stole it from Iryna's PR: https://github.com/Invoca/invoca_logging/pull/64/files :)

**Pull request merged**